pkg/relayer/cmd/cmd.go

```diff
@@ -202,6 +202,7 @@ func setupRelayerDependencies(
 		config.NewSupplyRingCacheFn(),
 		supplyTxFactory,
 		supplyTxContext,
+		// TODO_IN_THIS_PR: should we have many txclients - one for each key?
```

Comment:
@red-0ne what do you think we should do about proof/claim code? I've started going down the path of changing `supplierClient` and `txClient` to support multiple `signingKeyNames`, but that's a lot of changes and I'm not sure if that's the right thing to do. I'm thinking we can add an address to `sessionTree`, that way we know which client/supplier to reference when submitting transactions. Or maybe there should be multiple `sessionTree`s for each address? So many questions! :D 

Handling relays seems straightforward enough so far.

Reply:
Without having full context or diving into the code, IMO the best approach would be:

- supplierClientMap: map (signing address) -> supplierClient
- txClientMap: map (signing address) -> txClient

Seems like a good `SOLID` design.

Reply:
We definitely need to add the supplier address in the `SessionTree` struct [1]. We will then group messages by supplier when claiming [2] and proving [3]

We can have a `supplierClientMap` and a `txClientMap` and then match the C&P tx signers. This will involve changing what is supplied with `depinject` [4]. It will also create a websocket connection for each txClient.

Another approach would be to have multiple signers in the `SupplierClient` and `txClient` and will have to change the tx listener [5] to catch transactions for all the suppliers.

[1] https://github.com/pokt-network/poktroll/blob/main/pkg/relayer/session/sessiontree.go#L21
[2] https://github.com/pokt-network/poktroll/blob/main/pkg/relayer/session/claim.go#L156
[3] https://github.com/pokt-network/poktroll/blob/main/pkg/relayer/session/proof.go#L190
[4] https://github.com/pokt-network/poktroll/blob/main/pkg/relayer/cmd/cmd.go#L206-L207
[5] https://github.com/pokt-network/poktroll/blob/main/pkg/client/tx/client.go#L172

Reply:
> Another approach would be to have multiple signers in the SupplierClient and txClient and will have to change the tx listener [5] to catch transactions for all the suppliers.

👎 This doesn't follow good encapsulation principles IMO.

> We can have a supplierClientMap and a txClientMap and then match the C&P tx signers. This will involve changing what is supplied with depinject [4]. It will also create a websocket connection for each txClient.

👍  Seems simple, extendible, easier to debug, etc...

What @red-0ne said